spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RuntimeAppsController.java

```diff
@@ -99,24 +104,26 @@ public RuntimeAppsController(StreamDefinitionRepository streamDefinitionReposito
 	}
 
 	@RequestMapping
-	public PagedResources<AppStatusResource> list(PagedResourcesAssembler<AppStatus> assembler) {
-		List<AppStatus> values = new ArrayList<>();
+	public PagedResources<AppStatusResource> list(PagedResourcesAssembler<AppStatus> assembler) throws ExecutionException, InterruptedException {
+		List<StreamDefinition> asList = new ArrayList<>();
 		for (StreamDefinition streamDefinition : this.streamDefinitionRepository.findAll()) {
-			for (StreamAppDefinition streamAppDefinition : streamDefinition.getAppDefinitions()) {
-				String key = DeploymentKey.forStreamAppDefinition(streamAppDefinition);
-				String id = this.deploymentIdRepository.findOne(key);
-				if (id != null) {
-					values.add(appDeployer.status(id));
-				}
-			}
+			asList.add(streamDefinition);
 		}
-		Collections.sort(values, new Comparator<AppStatus>() {
-			@Override
-			public int compare(AppStatus o1, AppStatus o2) {
-				return o1.getDeploymentId().compareTo(o2.getDeploymentId());
-			}
-		});
-		return assembler.toResource(new PageImpl<>(values), statusAssembler);
+
+		List<AppStatus> statuses = forkJoinPool.submit(() ->
```

Comment:
@ericbottard Is [this](https://blog.krecan.net/2014/03/18/how-to-specify-thread-pool-for-java-8-parallel-streams/) the reason you are using ForkJoinPool + Parallel Stream approach?